### ISSUE TITLE
Unhandled promise rejection in stopspinner

### ISSUE THREAD

**Issue by mycroes** [zposten/unorepo#13]
Just tried out uno for the first time today, although basically it seems to work OK it does occasionally run into the following issue:

```
(node:31936) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'stop' of null
    at Object.stopSpinner (C:\Users\michaelc\AppData\Roaming\npm\node_modules\unorepo\dist\unorepo.js:2:2429)
    at C:\Users\michaelc\AppData\Roaming\npm\node_modules\unorepo\dist\unorepo.js:2:5064
    at s (C:\Users\michaelc\AppData\Roaming\npm\node_modules\unorepo\dist\unorepo.js:2:12200)
    at Generator._invoke (C:\Users\michaelc\AppData\Roaming\npm\node_modules\unorepo\dist\unorepo.js:2:11988)
    at Generator.t.<computed> [as next] (C:\Users\michaelc\AppData\Roaming\npm\node_modules\unorepo\dist\unorepo.js:2:12622)
    at n (C:\Users\michaelc\AppData\Roaming\npm\node_modules\unorepo\dist\unorepo.js:2:2640)
    at c (C:\Users\michaelc\AppData\Roaming\npm\node_modules\unorepo\dist\unorepo.js:2:2842)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:31936) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 29)
```

I also think the uno might be starting the passed script multiple times. This mainly happens when multiple files change, either by Save all in VS code or a git rebase.

The command I'm using: `uno watch --ignore **/*.d.ts,build --script build`.

**Comment by 0livare** [comment]
Hi @mycroes, thanks for the issue!

I have definitely seen git rebase mess up the watching.  I do want to fix that, but my solution to that for the moment is to just quit and re-run uno.

But I have never seen a problem with save all.  I've even mapped that command to ctrl+s so that it happens literally every time I save.

Are you sure you've seen this error on a run of uno where you haven't rebased?

Also, can you please provide your node, npm, and OS versions?

**Comment by mycroes** [comment]
Hi @zposten,

To be honsted, I'm not absolutely sure I've had it on _Save all_, I think we can safely assume that was misperception on my end. I (pointlessly) save all the time, so I don't use _Save all_ that often (I rebase more often than I use _Save all_)  and since that doesn't seem to be an issue for you, it probably isn't an issue for me. I'll try to pay attention, if it does happen I'll let you know.

About the version:
node: v13.2.0
npm: 6.13.7
OS: Windows 10 1903

For now I'm following the same solution, but it's annoying because I don't rebase from within VS code, so it doesn't show up directly. It also seems to create some lag when this happens, perhaps based on the size (number of affected files) of the rebase.

**Comment by 0livare** [comment]
Thanks for the information!  I'm working on a better strategy for change detection and debouncing in an effort to fix this issue

**Comment by 0livare** [comment]
@mycroes I just released v0.1.0 after merging #14.  I believe that should fix the problems you're having, please let me know if it does!

**Comment by mycroes** [comment]
I'd say this is mostly fixed (at least it's by far not as broken as it used to be). What I'm seeing now is that it does build twice on a rebase, not sure if that's to be expected. Output:

```
$ uno watch -i build,**/*.d.ts
[uno]  Watching the following packages:
[uno]    @vlc/matrix-visualization
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\BasicEndpoint.tsx
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Item\index.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\Endpoint.tsx
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\getRelativeEndpointItemRect.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\EndpointItemProps.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\EndpointItems.tsx
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\EndpointProps.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\AccessPoint\index.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\AccessPoint\getAccessPointTransform.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\AccessPoint\LiveAccessPoint.tsx
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\AccessPoint\AccessPointProps.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\AccessPoint\AccessPoint.tsx
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\VisualizationState\VisualizationState.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\index.ts
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Endpoint\LiveEndpoint.tsx
[uno]  CHANGE: C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\src\Visualization\Visualization.tsx
[uno]  $ yarn run build:cjs && yarn run build:esm && yarn run build:types && yarn run build:copy-files
[uno]  $ cross-env NODE_ENV=production BABEL_ENV=cjs babel --config-file ../../babel.config.js --extensions ".ts,.tsx" ./src --out-dir ./build --ignore "**/*.test.js"
[uno]  Successfully compiled 90 files with Babel.
[uno]  $ cross-env NODE_ENV=production BABEL_ENV=esm babel --config-file ../../babel.config.js --extensions ".ts,.tsx" ./src --out-dir ./build/esm --ignore "**/*.test.js"
[uno]  Successfully compiled 90 files with Babel.
[uno]  $ tsc --emitDeclarationOnly
[uno]  $ node ../../scripts/copy-files.js
[uno]  Created package.json in C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\build\package.json
[uno]    
[uno]  ✅  Successfully ran "yarn run build" in @vlc/matrix-visualization
[uno]
[uno]
[uno]
[uno]  ✅  Successfully ran "yarn run build" in 1 (of 1) packages
[uno]  ✅    @vlc/matrix-visualization
[uno]  $ yarn run build:cjs && yarn run build:esm && yarn run build:types && yarn run build:copy-files
[uno]  $ cross-env NODE_ENV=production BABEL_ENV=cjs babel --config-file ../../babel.config.js --extensions ".ts,.tsx" ./src --out-dir ./build --ignore "**/*.test.js"
[uno]  Successfully compiled 90 files with Babel.
[uno]  $ cross-env NODE_ENV=production BABEL_ENV=esm babel --config-file ../../babel.config.js --extensions ".ts,.tsx" ./src --out-dir ./build/esm --ignore "**/*.test.js"
[uno]  Successfully compiled 90 files with Babel.
[uno]  $ tsc --emitDeclarationOnly
[uno]  $ node ../../scripts/copy-files.js
[uno]  Created package.json in C:\Users\michaelc\Documents\Projects\VLC\Matrix.Next\js\visualization\build\package.json
[uno]    
[uno]  ✅  Successfully ran "yarn run build" in @vlc/matrix-visualization
[uno]
[uno]
[uno]
[uno]  ✅  Successfully ran "yarn run build" in 1 (of 1) packages
[uno]  ✅    @vlc/matrix-visualization
```

I can certainly live with this and I can even imagine this being intentional behavior (start a build on first change, schedule next build during cooldown?).
